### PR TITLE
[FEAT] `/gh me` Github Profile Feature with Share Profile and Issue Display

### LINKED ISSUE #30: GitHub Profile
## Why do we need this ?

As a GitHub App developer

**I need to**: 

Add a new GitHub Profile modal which can be triggered using `/gh me` or `/gh profile` command, which can be uses to fetch users details inside  Rocket.Chat

**So That**:

Users can share their profile, their own repositories, starred repositories etc to Rocket.Chat channels

## Acceptance Criteria

- [ ] Add new slash command which triggers the `UserProfileModal`
- [ ] Add  new `UserProfileModal` and supporting modals to view relevant data - repositories, activity, starred repos etc inside Rocket.Chat
- [ ] Add functionality to share the fetched results to Rocket.Chat channels similar to [GitHub Search and Share](https://github.com/RocketChat/Apps.Github22/pull/15#issuecomment-1213267447)
- [ ] Add the new function to fetch users data in the `GitHubSDK`

## Further Comments / References  

- Implementing share functionality : [GitHub Search and Share](https://github.com/RocketChat/Apps.Github22/pull/15#issuecomment-1213267447)
- [GitHub Rest APIs](https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api)

### PR DESCRIPTION
**Pull request by henit-chobisa**
## Issue(s) 

Closes #30 

## Screenshots
image
image
image
image

## Video : 

https://user-images.githubusercontent.com/72302948/204878155-d9a81953-c114-470b-82ff-be712cb0371a.mp4

## Acceptance Criteria fulfillment

- [x] Add new slash command which triggers the UserProfileModal
- [x] Add new UserProfileModal and supporting modals to view relevant data - repositories, activity, starred repos etc inside Rocket.Chat
- [x] Add functionality to share the fetched results to Rocket.Chat channels similar to https://github.com/RocketChat/Apps.Github22/pull/15#issuecomment-1213267447
- [x] Add the new function to fetch users data in the GitHubSDK

### CODE REVIEW

**Comment by henit-chobisa** [comment]
@samad-yar-khan
Hello Samad, I have made the desired changes, can you please review it and suggest me the changes, so that I can fix it as soon as possible.

**Comment by samad-yar-khan** [comment]
@henit-chobisa Thankyou for this PR. A much awaited feature 🙌. I will review the code changes and test it once and let you know.

**Comment by henit-chobisa** [comment]
@samad-yar-khan 
Sure, thank you Samad!

**Review by samad-yar-khan — COMMENTED** [review]
@henit-chobisa Everything looks great, just some minor changes. Please also remove all the code that is not being used at the moment as discussed in the last meeting.

File: github/handlers/UserProfileHandler.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    room: IRoom,
+    modify: IModify,
+    uikitcontext?: UIKitInteractionContext
+){
+    let access_token = await getAccessTokenForUser(
+        read,
+        context.getSender(),
+        app.oauth2Config
+    );
+    if (access_token && access_token.token){
```
Comment (samad-yar-khan) [inline github/handlers/UserProfileHandler.ts:25]: I have written similar statements in code but I think we can use `access_token?.token`  moving forward, makes it more readable.

File: github/lib/CreateIssueStatsBar.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                        ? OcticonIcons.ISSUE_OPEN
+                        : OcticonIcons.ISSUE_CLOSED,
+                altText: "State",
+            }),
+            block.newPlainTextObject(`${issueInfo.state}`),
+            block.newImageElement({
+                imageUrl: issueInfo.user_avatar ?? "",
+                altText: "User Image",
+            }),
+            block.newPlainTextObject(`Created by ${issueInfo.user_login}` ?? ""),
```
Comment (samad-yar-khan) [inline github/lib/CreateIssueStatsBar.ts:40]: @henit-chobisa  `??` operator is new for me, thanks  :) Are there any advantages to using this over  using simple `||` ? If there is no specific case where this helps, I think we can change it to `||` . Not a suggested change. Just curious

File: github/modals/UserProfileModal.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    block.addDividerBlock();
+
+    block.addImageBlock({imageUrl : `https://activity-graph.herokuapp.com/graph?username=${userInfo.username}&bg_color=ffffff&color=708090&line=24292e&point=24292e`, altText: "Github Contribution Graph"});
+
+
+
+    block.addDividerBlock();
+
+    block.addSectionBlock({
+        text: block.newPlainTextObject("Where should we teleport ?")
```
Comment (samad-yar-khan) [inline github/modals/UserProfileModal.ts:78]: Cool, but I think we can keep it simpler. `teleport` might be a heavy word for users without context. What do you think ?

File: github/modals/UserProfileModal.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    block.addContextBlock({
+        elements: [
+            block.newPlainTextObject(`followers: ${userInfo.followers}`),
+            block.newPlainTextObject(`following: ${userInfo.following}`)
+        ]
+    });
+
+    block.addDividerBlock();
+
+    block.addImageBlock({imageUrl : `https://activity-graph.herokuapp.com/graph?username=${userInfo.username}&bg_color=ffffff&color=708090&line=24292e&point=24292e`, altText: "Github Contribution Graph"});
```
Comment (samad-yar-khan) [inline github/modals/UserProfileModal.ts:71]: @henit-chobisa  I think we can make a simple method like `githubActivityGraphUrl(username)` to generate this `imageUrl` in `helpers`  so if something needs to be changed about this url and we are using it in more than one place, we can just update the method.

File: github/processors/bodyMarkdowmRenderer.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    }
+    catch(e){
+        console.log(e);
+    }
+
+    return text;
+}
+
+
+export async function BodyMarkdownRenderer({
```
Comment (samad-yar-khan) [inline github/processors/bodyMarkdowmRenderer.ts:23]: @henit-chobisa  No change, just wanted to thank you for working on this !! Really appreciate your work with the markdown rendering :clap: :clap: :smile:

**Review by samad-yar-khan — COMMENTED** [review]
@henit-chobisa Just a few more changes. :raised_hands:

File: github/processors/bodyMarkdowmRenderer.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                                body.length
+                            ))
+                        ),
+                    ],
+                });
+            }
+        });
+    }
+
+    // block.addSectionBlock({
```
[the remark this answers is not in the document]
Comment (henit-chobisa) [reply github/processors/bodyMarkdowmRenderer.ts:106]: Removed 🫡

File: github/handlers/UserProfileHandler.ts (hunk 1 above)
Comment (henit-chobisa) [reply github/handlers/UserProfileHandler.ts:25]: Sure Samad, Changed 🫡

File: github/lib/CreateIssueStatsBar.ts (hunk 2 above)
Comment (henit-chobisa) [reply github/lib/CreateIssueStatsBar.ts:40]: Ohh I see, I am a bit habitual to this one actually replacing null values using the succeeding string, 
Actually both the operators kinda work the same way, logically opposite of each other 🐶

File: github/modals/UserProfileModal.ts (hunk 3 above)
Comment (henit-chobisa) [reply github/modals/UserProfileModal.ts:78]: Sure Sure, I missed a chance to replace this keyword.

File: github/modals/UserProfileModal.ts (hunk 4 above)
Comment (henit-chobisa) [reply github/modals/UserProfileModal.ts:71]: Understood, I have declared the same method in helpers. ⌨️

File: github/processors/bodyMarkdowmRenderer.ts (hunk 5 above)
Comment (henit-chobisa) [reply github/processors/bodyMarkdowmRenderer.ts:23]: Thank you so much, Samad,
I love the concept of the "GitHub App" actually and I think if we work harder on this, we can make it stand in those amazing features of RC, that make customers helpless to switch from their existing platforms to RC. 😋

**Comment by samad-yar-khan** [comment]
@henit-chobisa  LGTM ! Thanks a lot for this PR :100:

**Comment by henit-chobisa** [comment]
> @henit-chobisa LGTM ! Thanks a lot for this PR 💯

My pleasure to contribute to such an amazing concept! 😊

**Pull request merged**